agent/connect/ca/provider_aws.go

```diff
@@ -603,7 +603,7 @@ func (a *AWSProvider) Sign(csr *x509.CertificateRequest) (string, error) {
 
 // SignIntermediate implements Provider
 func (a *AWSProvider) SignIntermediate(csr *x509.CertificateRequest) (string, error) {
-	err := validateSignIntermediate(csr, &connect.SpiffeIDSigning{ClusterID: a.clusterID, Domain: "consul"})
```

Comment:
The other two providers were using `connect.SpiffeIDSigningForCluster` instead of constructing a literal. I figured it would be good to have all 3 providers use this function to show that they do the same thing.